pkg/kubelet/images/image_gc_manager.go

```diff
@@ -237,8 +252,14 @@ func (im *realImageGCManager) detectImages(ctx context.Context, detectTime time.
 	// Make a set of images in use by containers.
 	for _, pod := range pods {
 		for _, container := range pod.Containers {
-			klog.V(5).InfoS("Container uses image", "pod", klog.KRef(pod.Namespace, pod.Name), "containerName", container.Name, "containerImage", container.Image, "imageID", container.ImageID)
-			imagesInUse.Insert(container.ImageID)
+			if !isRuntimeClassInImageCriAPIEnabled {
+				klog.V(5).InfoS("Container uses image", "pod", klog.KRef(pod.Namespace, pod.Name), "containerName", container.Name, "containerImage", container.Image, "imageID", container.ImageID)
+				imagesInUse.Insert(container.ImageID)
+			} else {
+				imageKey := getImageTuple(container.ImageID, container.ImageRuntimeHandler)
```

Comment:
right here I think we might be able to get away with finding the handler by looking up the pod runtime class

Reply:
Added response in previous comment thread: https://github.com/kubernetes/kubernetes/pull/121456#discussion_r1378100936

Reply:
https://github.com/kubernetes/kubernetes/pull/121456#discussion_r1378166991 @mikebrow please let me know if this is ok

Reply:
Discussed with @mikebrow on slack. This will be TODO for beta and I'll add this to sig-node meeting agenda for the next meeting and have a wider discussion with folks before updating the repo with these changes. thanks!

Reply:
track and change this to address scoping is pod in a follow up... this given the runtime manager where the runtime class manager is ... is not readily available to the image managers

Reply:
sgtm